File: src/frontend/src/ui/connectors/categories/ArchiveConnectorCategory.ts
import { type ConnectorCategoryID } from "@common/data/entities/connector/Connector";

import { ConnectorCategory } from "./ConnectorCategory";

export class ArchiveConnectorCategory extends ConnectorCategory {
    public static readonly CategoryID: ConnectorCategoryID = "archive";

    public constructor() {
        super("Archive", "Store your project data in a long-term external storage archive.");
    }
}

File: src/connectors/base/data/entities/connector/categories/archive_connector_category.py
import typing
from dataclasses import dataclass

from common.py.data.entities.connector import ConnectorCategoryID

from .connector_category import ConnectorCategory


@dataclass(frozen=True, kw_only=True)
class ArchiveConnectorCategory(ConnectorCategory):
    category_id: typing.ClassVar[ConnectorCategoryID] = "archive"

    def __init__(self):
        super().__init__(name="Archive")
